### PR TITLE
Add locations to Worker Activity Report

### PR DESCRIPTION
**Pull request by esoergel**
http://manage.dimagi.com/default.asp?170759
@sravfeyn

### CODE REVIEW

File: corehq/apps/reports/dont_use/fields.py [hunk 1]
```diff
@@ -260,16 +239,3 @@ class ReportMultiSelectField(ReportSelectField):
...
-    default_option = ['_all']
-    placeholder = 'Click to select groups'
-    help_text = "Start typing to select one or more groups"
-
-    @property
-    def options(self):
-        self.groups = Group.get_reporting_groups(self.domain)
-        opts = [dict(val=group.get_id, text=group.name) for group in self.groups]
-        opts.insert(0, {'text': 'All', 'val': '_all'})
-        return opts
```
Comment (esoergel) [inline corehq/apps/reports/dont_use/fields.py]: I needed to make some modifications to this filter, and I wasn't about to do it on the old `dont_use.fields` class, so I removed references to these old fields (not used anywhere else)

**Comment by sravfeyn** [comment]
Looks good

**Comment by sravfeyn** [comment]
Should we start adding unit tests to report filters and reports? I say this for following reasons.
- The obvious reason of ensuring that report displays what we intend to
- Better readibility/documentation for devs, to easily understand what a particular report displays
- To be confidently communicate to users how the filtering for each of the report works (and thereby tweak if there are new changes asked, like this)

I see some tests in that spirit [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_form_export.py) and [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_case_export.py). But we can incrementally start adding more tests to reports. 

Not sure if I am overlooking already existing tests or something, but this has been lingering thought for me for a while. Feel this is the right PR to raise these points. (not going to block the PR with this discussion anyway). Wanted to get others opinion, starting with yours @esoergel

**Comment by esoergel** [comment]
Yeah, we should really test these.  It's pretty difficult to do though, as there's a whole lot which needs to be mocked before it's possible to query these things.  A lot of it could probably be pulled out in to helper functions and tested there though.

I'm actually going to hold off on this PR for a bit though, I don't think showing just information per location is acceptable - users would expect to see stuff for a location and all their children.  I spoke with Mealz and Sheelz and they agree.

**Pull request closed without merging**